src/devices/src/virtio/balloon/persist.rs

```diff
@@ -20,12 +20,92 @@ use crate::virtio::balloon::device::{BalloonStats, ConfigSpace};
 use crate::virtio::persist::VirtioDeviceState;
 use crate::virtio::{DeviceState, Queue};
 
-#[derive(Versionize)]
+#[derive(Clone, Versionize)]
+pub struct BalloonConfigSpaceState {
+    num_pages: u32,
+    actual_pages: u32,
+}
+
+impl BalloonConfigSpaceState {
+    fn from_config_space(config_space: ConfigSpace) -> Self {
+        Self {
+            num_pages: config_space.num_pages,
+            actual_pages: config_space.actual_pages,
+        }
+    }
+
+    fn create_config_space(&self) -> ConfigSpace {
+        ConfigSpace {
+            num_pages: self.num_pages,
+            actual_pages: self.actual_pages,
+        }
+    }
+}
+
+#[derive(Clone, Versionize)]
+pub struct BalloonStatsState {
+    target_pages: u32,
+    actual_pages: u32,
+    target_mb: u32,
+    actual_mb: u32,
```

Comment:
Are the target_* fields part of the stats or user input ?

Reply:
Also from a state snapshot point of view, actual_pages/mb seems to be duplicated in config space.

Reply:
Removed them as they are populated when a GET "/balloon/statistics" anyway.